### ISSUE TITLE
Open to refactoring to support geo-based selection of email target?

### ISSUE THREAD

**Issue by JoeMurray** [progressivetech/com.aghstrategies.petitionemail#6]
One feature commonly seen with email petition campaigns is to have the person signing the petition contact their local representative. The local representative can be determined by geo-coded lookup, sometimes lat/long based, sometimes postal code based. The official might be a federal, provincial/state, or local municipal official. There are various services that help do these sorts of lookups and provide contact information for the representatives, e.g. for Canada https://represent.opennorth.ca/api/

I think it would be reasonable to have one or more extensions that are responsible for retrieving information from these external services (and probably caching the data in CiviCRM tables). And then this extension could if they are present enable options for determining the target of a petition campaign, e.g. federal officials.

Would @jmcclelland and/or @mattwire be comfortable with accepting contributions for that from JMA?

**Comment by hesco** [comment]
Thank you, JoeMurray:

This would be a most appreciated addition, that is if I could ever get a successful installation.  
See:  https://github.com/progressivetech/com.aghstrategies.petitionemail/issues/5

**Comment by jmcclelland** [comment]
Yes, this feature is currently available via the [electoral extension](https://github.com/progressivetech/com.jlacey.electoral). You _could_ submit a PR to add an extra source directly to this extension, but better, you can submit PR against the Electoral extension instead and this extension will automatically pick it up.

**Comment by jmcclelland** [comment]
See [dynamic targets](https://github.com/progressivetech/com.aghstrategies.petitionemail#dynamic-targets).
